weblate/trans/models/unit.py

```diff
@@ -899,11 +902,22 @@ def run_checks(self, same_state=True, same_content=True, is_new=False):
             self.source_checks().values_list('check', flat=True)
         )
 
+        check_pair = set(FLAGS_PAIR_RE.findall('\n'.join(self.all_flags)))
+
         # Run all checks
         for check in checks_to_run:
             check_obj = CHECKS[check]
+
+            if len(check_pair) > 0 and hasattr(check_obj, 'check_value'):
+                check_value = max(
+                    {(x) for x in check_pair if x[0] == check},
+                    key=lambda v: v[1]
+                )[1]
+                setattr(check_obj, 'check_value', check_value)
```

Comment:
> I'd rather do the parsing in the check itself....

Could you tell me more details? I think I missed something and I haven't realized yet.

For me, the problem is that given flag string is like 'max-length:100' - key:value. To load check object by name, I tried to extract  key part from the string and the key part is used for loading a proper check object.

To make check object parse itself, each object should keep key:value string as parameter or attribute. If so, as I understand, it will be needed to create the 'max-length' check object per value because one check object can handle one parameter. e.g, "max-lengh:50" and "max-length:100",  might be different objects.